### PR TITLE
Add zeitwerk to handle autoload dependencies

### PR DESCRIPTION
**Pull request by valter0ff**
Using Zeitwerk gem to handle autoload required files

### CODE REVIEW

**Comment by ronaldtse** [comment]
@zznull @valter0ff thank you for the PR! The tests are currently failing since zeitwerk 2.7.1 requires Ruby 3.2, but we need to support Ruby 3.0. Can we use a zeitwerk version that supports Ruby 3.0?

**Comment by valter0ff** [comment]
> @zznull @valter0ff thank you for the PR! The tests are currently failing since zeitwerk 2.7.1 requires Ruby 3.2, but we need to support Ruby 3.0. Can we use a zeitwerk version that supports Ruby 3.0?

Yes, Zeitwerk 2.6 supports ruby from 2.5. But I figured out some issue with running rspec, working on it

File: niso-jats.gemspec [hunk 1]
```diff
@@ -33,7 +33,7 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency "lutaml-model"
   spec.add_runtime_dependency "nokogiri"
-  spec.add_dependency "zeitwerk", "~> 2.7"
+  spec.add_runtime_dependency "zeitwerk", "~> 2.6"
```
Comment (Novtopro) [inline niso-jats.gemspec:36]: Should be something like `~> 2.6.18`

**Review by ronaldtse — APPROVED** [review]

File: niso-jats.gemspec (hunk 1 above)
Comment (ronaldtse) [reply niso-jats.gemspec:36]: Agree with @Novtopro here, as this lines means zeitwerk "2.7" which is Ruby 3.2+ is also applicable.

**Comment by ronaldtse** [comment]
Thank you @valter0ff ! This is now merged.

**Pull request merged**